src/web/components/SharingPermission/SharingPermissionsTable.tsx

```diff
@@ -21,64 +24,69 @@ function NoParticipant() {
     </div>
   );
 }
-export function SharingPermissionsTable({ sharedParticipants }: SharingPermissionsTableProps) {
+export function SharingPermissionsTable({
+  sharedParticipants,
+  onDeleteSharingPermission,
+  children,
+}: SharingPermissionsTableProps) {
   const [filterText, setFilterText] = useState('');
-  const [checkedParticipants, setCheckedParticipants] = useState<number[]>([]);
-  const [selectAll, setSelectAll] = useState(false);
+  const [checkedParticipants, setCheckedParticipants] = useState<Set<number>>(new Set());
+  const hasParticipantSelected = useMemo(() => checkedParticipants.size > 0, [checkedParticipants]);
```

Comment:
Do you need a memo here? Checking `.size>0` is a very cheap operation, and the overhead of the memo is probably much higher.